paddle/fluid/framework/block_desc.cc

```diff
@@ -46,11 +46,25 @@ VarDesc *BlockDesc::FindVarRecursive(const std::string &name) const {
   if (name == kEmptyVarName) return nullptr;
 
   auto it = vars_.find(name);
-  if (it == vars_.end()) {
-    return Parent() == kNoneBlockIndex ? nullptr
-                                       : ParentBlock()->FindVarRecursive(name);
+  if (it != vars_.end()) {
+    return it->second.get();
   }
-  return it->second.get();
+
+  BlockDesc *tmp = ParentBlock();
```

Comment:
In issue #8367, the search is described as BFS. But the code here is DFS. Please confirm.